sdks/python/apache_beam/typehints/typehints_test.py

```diff
@@ -1062,57 +797,92 @@ def test_hint_helper(self):
     self.assertTrue(is_consistent_with(str, Union[str, int]))
     self.assertFalse(is_consistent_with(Union[str, int], str))
 
-  def test_positional_arg_hints(self):
-    self.assertEqual(typehints.Any, _positional_arg_hints('x', {}))
-    self.assertEqual(int, _positional_arg_hints('x', {'x': int}))
-    self.assertEqual(typehints.Tuple[int, typehints.Any],
-                     _positional_arg_hints(['x', 'y'], {'x': int}))
-
   def test_getcallargs_forhints(self):
     def func(a, b_c, *d):
       b, c = b_c # pylint: disable=unused-variable
       return None
     self.assertEqual(
         {'a': Any, 'b_c': Any, 'd': Tuple[Any, ...]},
-        getcallargs_forhints(func, *[Any, Any]))
+        getcallargs_forhints(False, func, *[Any, Any]))
+    if sys.version_info >= (3,):
+      self.assertEqual(
+          {'a': Any, 'b_c': Any, 'd': Tuple[Union[int, str], ...]},
+          getcallargs_forhints(False, func, *[Any, Any, str, int]))
+    else:
+      self.assertEqual(
+          {'a': Any, 'b_c': Any, 'd': Tuple[Any, ...]},
+          getcallargs_forhints(False, func, *[Any, Any, Any, int]))
```

Comment:
Why are we testing something different here (and below)?

Reply:
The Python 2 version (`getcallargs_forhints_impl_py2`) has a bug in it that always sets varargs to `Tuple[Any, ...]`. I left in a TODO since I didn't want to complicate this PR any further.